src/pages/Search/components/Heatmap.js

```diff
@@ -12,36 +16,56 @@ import {
   BoxRow,
   TimeZone,
 } from './Heatmap.styles';
-import Box from './Box';
 
 const Heatmap = ({ posts }) => {
-  const rows = posts.map((el) => {
-    const columns = el.map((totalPosts) => (
-      <Box numTotalPosts={totalPosts.length} />
-    ));
-    return <BoxRow key={el}>{columns}</BoxRow>;
+  const [selected, setSelected] = useState('');
+  const [timePosts, setTimePosts] = useState([]);
+
+  const handleActive = (key) => setSelected(key);
+
+  const setPosts = (selectedPosts) => setTimePosts(selectedPosts);
+
+  const rows = posts.map((el, elIndex) => {
+    const columns = el.map((totalPosts, i) => {
+      const { length } = totalPosts;
+      const key = `${elIndex}-${length}-${i}`;
+      return (
+        <Box
+          key={key}
+          index={key}
+          selected={selected === key}
+          handleActive={handleActive}
+          setPosts={setPosts}
+          posts={totalPosts}
+        />
+      );
+    });
+    return <BoxRow key={uuid()}>{columns}</BoxRow>;
```

Comment:
Here it's ok to use `elIndex`, you can disable the ESLint warning for this file. It's important to not use `index` as key when the order of the elements can change. But in this case the weekdays are not dynamic.

Btw using uuid may be counterproductive here since every row will have a different key in each render cycle. Thus some internal React optimizations won't work I guess

Reply:
should I use an index as key for `<BoxRow />`

Reply:
Yes, in this case elIndex is fine to use. You can also use the name of the day if you like, but it's effectively the same